godplugin/plugin.go

```diff
@@ -47,6 +48,43 @@ func New() *Plugin {
 	return &Plugin{
 		modules: make(modules.Registry),
 		checkCh: make(chan Job, 1),
+		loopQueue: &jobQueue{
+			mux: &sync.Mutex{},
+		},
+	}
+}
+
+type jobQueue struct {
+	mux   *sync.Mutex
+	queue []Job
+}
+
+func (q *jobQueue) add(job Job) {
+	q.mux.Lock()
+	defer q.mux.Unlock()
+
+	q.queue = append(q.queue, job)
+}
+
+func (q *jobQueue) pop(fullName string) Job {
```

Comment:
pop is not a good name because it means remove last item and return that

Reply:
agreed